CharlieBackend.Api.UnitTes/RepositoriesTests/DashboardRepositoryTest.cs

```diff
@@ -0,0 +1,744 @@
+                new StudentMarkDto
+                {
+                    CourseId = 2,
+                    StudentGroupId = 1,
+                    StudentId = 12,
+                    LessonId = 10,
+                    LessonDate = new DateTime(),
+                    StudentMark = 90,
+                }
+            };
+
+            var mock = GetMockDbSetOfVisits();
+
+            _applicationContextMock.SetupGet(x => x.Visits).Returns(mock.Object);
+
+            var dashboardRepository = new DashboardRepository(_applicationContextMock.Object);
+
+            //Act
+
+            var result = await dashboardRepository.GetStudentsMarksListByStudentIds(new List<long> { 10, 12 });
+
+            //Assert
+
+            Assert.NotNull(result);
+            Assert.NotEmpty(result);
+            Assert.Equal(expected.Count, result.Count);
+            for (int i = 0; i < expected.Count; i++)
+            {
+                Assert.Equal(expected[i].ToString(), result[i].ToString());
+            }
+        }
+
+        [Fact]
+        public async Task GetStudentAverageMarksByStudentIdsAndGropsIdsAsync()
+        {
+            //Arrange
+
+            var expected = new List<AverageStudentMarkDto>
+            {
+                new AverageStudentMarkDto
+                {
+                    CourseId =2,
+                    StudentGroupId =1,
+                    StudentId =11,
+                    StudentAverageMark = 78
+                },
+                new AverageStudentMarkDto
+                {
+                    CourseId =2,
+                    StudentGroupId =1,
+                    StudentId =12,
+                    StudentAverageMark = 60
+                }
+            };
+
+            var mock = GetMockDbSetOfVisits();
+
+            _applicationContextMock.Setup(x => x.Visits).Returns(mock.Object);
+
+            var dashboardRepository = new DashboardRepository(_applicationContextMock.Object);
+
+            //Act
+
+            var result = await dashboardRepository.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(new List<long> { 11, 12 }, new List<long> { 1 });
+
+            //Assert
+
+            Assert.NotNull(result);
+            Assert.NotEmpty(result);
+            Assert.Equal(expected.Count, result.Count);
+            for (int i = 0; i < expected.Count; i++)
+            {
+                Assert.Equal(expected[i].ToString(), result[i].ToString());
+            }
+        }
+
+        [Fact]
+        public async Task GetStudentAverageVisitsPercentageByStudentIdsAsync()
+        {
+            //Arrange
+
+            var expected = new List<AverageStudentVisitsDto>()
+            {
+                new AverageStudentVisitsDto
+                {
+                    CourseId =2,
+                    StudentId =12,
+                    StudentGroupId = 1,
+                    StudentAverageVisitsPercentage = 50
+                }
+            };
+
+            var mock = GetMockDbSetOfVisits();
```

Comment:
Also it is a good practice to keep tests separated and don't share data setup. Usually, unit test is the first place to understand the code, so it is more convenient to keep all incoming data in the test.
For example in this test it is enough to create one student and two visits.